python/pyarrow/parquet/core.py

```diff
@@ -1887,10 +1887,23 @@ def read_table(source, *, columns=None, use_threads=True,
                 "the 'schema' argument is not supported when the "
                 "pyarrow.dataset module is not available"
             )
+        if isinstance(source, list):
+            raise ValueError(
+                "the 'source' argument cannot be a list of files "
+                "when the pyarrow.dataset is not available"
+            )
+
         filesystem, path = _resolve_filesystem_and_path(source, filesystem)
         if filesystem is not None:
-            source = filesystem.open_input_file(path)
-        # TODO test that source is not a directory or a list
+            try:
+                source = filesystem.open_input_file(path)
+            except (OSError, FileNotFoundError) as e:
```

Comment:
Will `filesystem.open_input_file` raise `OSError` and/or `FileNotFoundError` only when source is not a file and is a dir? Could for example raise `OSError` when the process doesn't have permission to read the file or others? 
If one of those exceptions can raise on other cases, we might be misleading users as we will always raise a `ValueError` with the description pointing to the dir vs file issue.